lib/SimpleSAML/Metadata/SAMLParser.php

```diff
@@ -1118,13 +1115,12 @@ private static function processExtensions($element, $parentExtensions = [])
                     $ret['UIInfo']['PrivacyStatementURL'] = $e->getPrivacyStatementURL();
 
                     foreach ($e->getKeywords() as $uiItem) {
-                        if (!($uiItem instanceof \SAML2\XML\mdui\Keywords)
-                            || ($uiItem->getKeywords() === [])
-                            || ($uiItem->getLanguage() === null)
-                        ) {
+                        $keywords = $uiItem->getKeywords();
```

Comment:
Is `$uiItem` guaranteed here to be an instance of `\SAML2\XML\mdui\Keywords`?

Reply:
Yes it is:
https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdui/UIInfo.php#L59
https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdui/UIInfo.php#L104

Reply:
In theory, it's not, as you can set the `Keywords` property there to an array of whatever objects. Look at [this](https://github.com/simplesamlphp/saml2/blob/release-3.x/src/SAML2/XML/mdui/UIInfo.php#L112-L120). However, in this case I think it's safe, since the property would only get populated with `\SAML2\XML\mdui\Keywords` objects via the constructor 👍

Reply:
We could add a Webmozart-assertion to the saml2-library to catch that?

Reply:
I think it's a good idea!